cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,8 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = listeners == null ? "localhost:9092" : listeners.substring(listeners.lastIndexOf("/") + 1, listeners.length());
```

Comment:
Note that `listeners` config can be specified without a `host` -- such as `SSL://:9091` (see https://kafka.apache.org/documentation/#listeners). However, providing an empty hostname in `bootstrap.servers` config will cause an exception. 

Should we limit this parsing to retrieval of `port`?

Reply:
Thanks  for your answer, you are right.Let me make some changes.